docs/Blog-Post.md

```diff
@@ -0,0 +1,276 @@
+hope that these libraries will be useful to the community, and we look forward to seeing what people do with them.
+
+## The Landscape of Foundation Model Training Frameworks
+
+Numerous foundation model training frameworks exist in the community, each with its strengths and focus.
+For large language models (LLMs) (the focus of this release),
+the most well-known in the open source community is probably NVIDIA's PyTorch-based [Megatron-LM](https://github.com/NVIDIA/Megatron-LM),
+and its many derivatives, including EleutherAI's [GPT-NeoX](https://github.com/EleutherAI/gpt-neox) codebase. MosaicML
+has recently released [LLM Foundry](https://github.com/mosaicml/llm-foundry), which we used to train
+[BioMedLM](https://crfm.stanford.edu/2022/12/15/biomedlm.html) last autumn.
+
+In the Jax community, there are a number of libraries popping up. Google has released [T5X](https://github.com/google-research/t5x)
+and [MaxText](https://github.com/google/maxtext). There are also a number of independent libraries, including [EasyLM](https://github.com/young-geng/EasyLM)
+and [JaxSeq](https://github.com/Sea-Snell/JAXSeq), both of which are based on [Flax](https://github.com/google/flax/)
+and modified libraries from [Hugging Face Transformers](https://github.com/huggingface/transformers/). Salesforce has
+released the Haiku-based [Jaxformer](https://github.com/salesforce/jaxformer).
+
+Despite the wide array of existing frameworks, when we started, we found that none of them fully addressed our needs.
+At CRFM, we wanted to achieve four goals:
+
+* **Legibility**: We wanted to be able to write code that was easy to read and understand, and that could be composed easily.
+* **Reproducibility**: We wanted to be able to reproduce our results exactly, even in the presence of preemption and
+  restarting from checkpoints.
+* **Scalability**: We wanted to be able to fully utilize the Cloud TPU resources we were given as part of Google's TPU Research Cloud program, as well as our own GPU resources.
+* **Efficiency**: We wanted to achieve the other three goals without sacrificing (much) efficiency.
+
+We chose [Jax](https://github.com/google/jax/) as our framework because it is a powerful, flexible, and performant,
+and offers strong reproducibility guarantees. Jax works well on TPUs, while we found that PyTorch support was still uneven.
+Jax is also a natural choice because it allows you to focus on the "what" of your code, and not on the "how:" details of
+partitioning and communication can be left to the XLA compiler. Finally, Jax makes reproducibility easy, since it uses
+bitwise deterministic PRNGs by default, with careful control over the PRNG state.
+
+However, Jax is a low-level framework, and we found that, by itself, it did not provide the legibility that we wanted.
+
+For us, legibility is a top concern. We therefore created two new libraries: **Haliax** and **Levanter**. Haliax is a
```

Comment:
Maybe should say relationship of haliax to flax?

Reply:
to Equinox? It has no relationship to Flax.

Reply:
I meant more as a related work since it seems like people will probably think of jax/flax, and we have jax/equinox/haliax...